File: two-pointer/905-Sort-Array-By-Parity.go
func sortArrayByParity(nums []int) []int {
    if len(nums) == 1 {
        return nums
    }

    left, right := 0, 1
    for left <= right && right < len(nums) {
        if nums[left] % 2 == 0 {
            left++
        }
        if nums[left] % 2 != 0 && nums[right] % 2 == 0 {
            nums[left], nums[right] = nums[right], nums[left]
            left++
        }
        right++
    }
    return nums
}

File: two-pointer/905-Sort-Array-By-Parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        if len(nums) == 1:
            return nums
        
        low, high = 0, 1
        while low <= high < len(nums):
            if nums[low] % 2 == 0:
                low += 1
            if nums[low] % 2 != 0 and nums[high] % 2 == 0:
                nums[low], nums[high] = nums[high], nums[low]
                low += 1
            high += 1
        return nums